src/overlays/actors/ovl_En_Gm/z_en_gm.c

```diff
@@ -20,37 +41,280 @@ const ActorInit En_Gm_InitVars = {
+    }
+}
+
+void EnGm_SetTextID(EnGm* this) {
+    switch (func_80A3D7C8()) {
+        case 0:
+            if (gSaveContext.infTable[11] & 1) {
+                this->actor.textId = 0x304B;
+            } else {
+                this->actor.textId = 0x304A;
+            }
+            break;
+        case 1:
+            if (gSaveContext.infTable[11] & 2) {
+                this->actor.textId = 0x304F;
+            } else {
+                this->actor.textId = 0x304C;
+            }
+            break;
+        case 2:
+            this->actor.textId = 0x304E;
+            break;
+        case 3:
+            this->actor.textId = 0x304D;
+            break;
+    }
+}
+
+void func_80A3DB04(EnGm* this, GlobalContext* globalCtx) {
+    f32 dx;
+    f32 dz;
+    Player* player = PLAYER;
+
+    dx = this->talkPos.x - player->actor.posRot.pos.x;
+    dz = this->talkPos.z - player->actor.posRot.pos.z;
+
+    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+        EnGm_SetTextID(this);
+        this->subActionFunc = func_80A3DC44;
+    } else if (func_8002F194(&this->actor, globalCtx)) {
+        this->subActionFunc = func_80A3DBF4;
+    } else if (this->collider.base.maskA & 2 || (SQ(dx) + SQ(dz)) < SQ(100.0f)) {
+        this->collider.base.acFlags &= ~2;
+        func_8002F2CC(&this->actor, globalCtx, 415.0f);
+    }
+}
+
+void func_80A3DBF4(EnGm* this, GlobalContext* globalCtx) {
+    if (func_8010BDBC(&globalCtx->msgCtx) == 6 && func_80106BC8(globalCtx)) {
+        this->subActionFunc = func_80A3DB04;
+    }
+}
+
+void func_80A3DC44(EnGm* this, GlobalContext* globalCtx) {
+    f32 dx;
+    f32 dz;
+    s32 pad;
+    Player* player = PLAYER;
+
+    EnGm_SetTextID(this);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Gm/func_80A3D7C8.s")
+    dx = this->talkPos.x - player->actor.posRot.pos.x;
+    dz = this->talkPos.z - player->actor.posRot.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Gm/func_80A3D838.s")
+    if (func_8002F194(&this->actor, globalCtx)) {
+        switch (func_80A3D7C8()) {
+            case 0:
+                gSaveContext.infTable[11] |= 1;
+            case 3:
+                this->subActionFunc = func_80A3DD7C;
+                return;
+            case 1:
+                gSaveContext.infTable[11] |= 2;
+            case 2:
+                this->subActionFunc = EnGm_ProcessChoiceIndex;
+            default:
+                return;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Gm/func_80A3D9D8.s")
+        this->subActionFunc = EnGm_ProcessChoiceIndex;
+    }
+    if (this->collider.base.maskA & 2 || (SQ(dx) + SQ(dz)) < 10000.0f) {
+        this->collider.base.acFlags &= ~2;
+        func_8002F2CC(&this->actor, globalCtx, 415.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Gm/func_80A3DA50.s")
+void func_80A3DD7C(EnGm* this, GlobalContext* globalCtx) {
+    u8 flag = func_8010BDBC(&globalCtx->msgCtx);
```

Comment:
We usually call that `dialogState`, tho that will probably change later but it's better than `flag` for now imo
```suggestion
    u8 dialogState = func_8010BDBC(&globalCtx->msgCtx);
```